src/Badge/Badge.component.tsx

```diff
@@ -2,45 +2,148 @@
 import * as React from 'react';
 // VENDOR
 import classNames from 'classnames';
-import styled from 'styled-components';
+import styled, { css, FlattenSimpleInterpolation } from 'styled-components';
 // THEME
-import { fonts, colors, sizes, variables } from '../theme';
+import { colors, fonts } from '../theme';
 
-export interface BadgeProps {
-    count?: string | number;
-    className?: string;
+interface BadgeProps {
+    backgroundColor?: string;
+    borderColor?: string;
+    borderColorHover?: string;
     children?: any;
+    className?: string;
+    count?: number;
+    isParentHovered?: boolean;
+    overflowCount?: number;
+    offsetBottom?: number;
+    offsetLeft?: number;
+    size?: 'dot' | 'small' | 'large';
+    standalone?: boolean;
+    textColor?: string;
+}
+
+interface BadgeStylesGroup {
+    [key: string]: FlattenSimpleInterpolation;
 }
 
+const BadgeSizeStyles: BadgeStylesGroup = {
+    dot: css`
+        height: 0.75rem;
+        width: 0.75rem;
+    `,
+    small: css`
+        font-size: 0.625rem;
+        height: 1.25rem;
+        min-width: 1.25rem;
+        padding: 0 0.25rem;
+    `,
+    large: css`
+        font-size: 0.75rem;
+        height: 1.75rem;
+        min-width: 1.75rem;
+        padding: 0 0.5rem;
+    `,
+};
+
 const StyledBadgeContainer = styled.div`
     position: relative;
     font-family: ${fonts.fontFamily};
     display: inline-block;
 `;
 
-const StyledBadge = styled.div`
-    position: absolute;
-    top: -0.5rem;
-    right: -0.25rem;
-    background-color: ${colors.flashPink.base};
-    color: ${colors.white.base};
+const StyledBadge = styled.div<BadgeProps>`
+    align-items: center;
+    background-color: ${({ backgroundColor }: BadgeProps) => backgroundColor};
+    border: 0.125rem solid ${({
+        borderColor,
+        borderColorHover,
+        isParentHovered,
+    }: BadgeProps) => (isParentHovered ? borderColorHover : borderColor)};
+    box-sizing: border-box;
+    color: ${({ textColor }: BadgeProps) => textColor};
+    display: flex;
     font-weight: 700;
-    font-size: 0.75rem;
-    padding: 1px ${sizes.padding.xs};
-    border-radius: ${variables.borderRadius};
+    justify-content: center;
+    text-align: center;
+
+    /* Border Radius */
+    ${({ count = 0, size = 'dot' }: BadgeProps) =>
```

Comment:
The linter is saying these lines are uncovered:
```
 70,76,86 
```

I'm not sure how to test those. Any ideas?

Reply:
Coverage reports all have their anomalies. Likely it'd be more difficult to test this than the value that would be offered by the test. I typically focus on gaining coverage of other uncovered JS files instead of trying to endlessly chase a few lines that I'm pretty confident in. In order to test these you'd need to mock both forks of the logic in your ternary inside of a spec.

Reply:
@sharkparty that makes sense. I think I have tests for the forks of the ternary, just not the assignment of the defaults. Could you remind me what the longhand of this is? 
```
 ${({ count = 0, size = 'dot' }: BadgeProps) =>
```

Reply:
@sharkparty that makes sense. I think I have tests for the forks of the ternary, just not the assignment of the defaults. Could you remind me what the longhand of this is?

 ${({ count = 0, size = 'dot' }: BadgeProps) =>